src/analysis/special_functions/pow.lean

```diff
@@ -567,3 +567,305 @@ lemma filter.tendsto.nnrpow {α : Type*} {f : filter α} {u : α → nnreal} {v
+    { have hx' : x ≠ 0, by simpa using hx,
+      have hy' : y ≠ 0, by simpa using hy,
+      simp only [some_eq_coe],
+      rw [← coe_mul, coe_rpow_of_ne_zero, nnreal.mul_rpow, coe_mul,
+          coe_rpow_of_ne_zero hx', coe_rpow_of_ne_zero hy'],
+      simp [hx', hy'] } }
+end
+
+lemma one_le_rpow {x : ennreal} {z : ℝ} (h : 1 ≤ x) (h₁ : 0 ≤ z) : 1 ≤ x^z :=
+begin
+  cases x,
+  { rcases le_iff_eq_or_lt.1 h₁ with H|H,
+    { simp [← H, le_refl] },
+    { simp [top_rpow_of_pos H] } },
+  { simp only [one_le_coe_iff, some_eq_coe] at h,
+    simp [coe_rpow_of_nonneg _ h₁, nnreal.one_le_rpow h h₁] }
+end
+
+lemma rpow_le_rpow {x y : ennreal} {z: ℝ} (h₁ : x ≤ y) (h₂ : 0 ≤ z) : x^z ≤ y^z :=
+begin
+  rcases le_iff_eq_or_lt.1 h₂ with H|H, { simp [← H, le_refl] },
+  cases y, { simp [top_rpow_of_pos H] },
+  cases x, { exact (not_top_le_coe h₁).elim },
+  simp at h₁,
+  simp [coe_rpow_of_nonneg _ h₂, nnreal.rpow_le_rpow h₁ h₂]
+end
+
+lemma rpow_lt_rpow {x y : ennreal} {z: ℝ} (h₁ : x < y) (h₂ : 0 < z) : x^z < y^z :=
+begin
+  cases x, { exact (not_top_lt h₁).elim },
+  cases y, { simp [top_rpow_of_pos h₂, coe_rpow_of_nonneg _ (le_of_lt h₂)] },
+  simp at h₁,
+  simp [coe_rpow_of_nonneg _ (le_of_lt h₂), nnreal.rpow_lt_rpow h₁ h₂]
+end
+
+lemma rpow_lt_rpow_of_exponent_lt {x : ennreal} {y z : ℝ} (hx : 1 < x) (hx' : x ≠ ⊤) (hyz : y < z) :
+  x^y < x^z :=
+begin
+  lift x to nnreal using hx',
+  rw [one_lt_coe_iff] at hx,
+  simp [coe_rpow_of_ne_zero (ne_of_gt (lt_trans zero_lt_one hx)),
+        nnreal.rpow_lt_rpow_of_exponent_lt hx hyz]
+end
+
+lemma rpow_le_rpow_of_exponent_le {x : ennreal} {y z : ℝ} (hx : 1 ≤ x) (hyz : y ≤ z) : x^y ≤ x^z :=
+begin
+  cases x,
+  { rcases lt_trichotomy y 0 with Hy|Hy|Hy;
+    rcases lt_trichotomy z 0 with Hz|Hz|Hz;
+    simp [Hy, Hz, top_rpow_of_neg, top_rpow_of_pos, le_refl];
+    linarith },
+  { simp only [one_le_coe_iff, some_eq_coe] at hx,
+    simp [coe_rpow_of_ne_zero (ne_of_gt (lt_of_lt_of_le zero_lt_one hx)),
+          nnreal.rpow_le_rpow_of_exponent_le hx hyz] }
+end
+
+lemma rpow_lt_rpow_of_exponent_gt {x : ennreal} {y z : ℝ} (hx0 : 0 < x) (hx1 : x < 1) (hyz : z < y) :
+  x^y < x^z :=
+begin
+  lift x to nnreal using ne_of_lt (lt_of_lt_of_le hx1 le_top),
+  simp at hx0 hx1,
+  simp [coe_rpow_of_ne_zero (ne_of_gt hx0), nnreal.rpow_lt_rpow_of_exponent_gt hx0 hx1 hyz]
+end
+
+lemma rpow_le_rpow_of_exponent_ge {x : ennreal} {y z : ℝ} (hx1 : x ≤ 1) (hyz : z ≤ y) :
+  x^y ≤ x^z :=
+begin
+  lift x to nnreal using ne_of_lt (lt_of_le_of_lt hx1 coe_lt_top),
+  by_cases h : x = 0,
+  { rcases lt_trichotomy y 0 with Hy|Hy|Hy;
+    rcases lt_trichotomy z 0 with Hz|Hz|Hz;
+    simp [Hy, Hz, h, zero_rpow_of_neg, zero_rpow_of_pos, le_refl];
+    linarith },
+  { simp at hx1,
+    simp [coe_rpow_of_ne_zero h,
+          nnreal.rpow_le_rpow_of_exponent_ge (bot_lt_iff_ne_bot.mpr h) hx1 hyz] }
+end
+
+lemma rpow_le_one {x : ennreal} {e : ℝ} (he : 0 ≤ e) (hx2 : x ≤ 1) : x^e ≤ 1 :=
```

Comment:
Should the order of hypotheses be made consistent with `one_le_rpow`, `one_lt_rpow` and `rpow_lt_one`? (That is, hypothesis on the base first, and then on the exponent.)

Ah, it looks like `nnreal.rpow_le_one` has the hypotheses flipped as well.

Reply:
Good catch. I have fixed this lemma and the other corresponding ones in the file.